source/server/admin/stats_handler.cc

```diff
@@ -132,13 +134,19 @@ Http::Code StatsHandler::handlerPrometheusStats(absl::string_view path_and_query
   const Http::Utility::QueryParams params =
       Http::Utility::parseAndDecodeQueryString(path_and_query);
   const bool used_only = params.find("usedonly") != params.end();
+  const bool text_readouts = params.find("text_readouts") != params.end();
+
+  absl::optional<std::vector<Stats::TextReadoutSharedPtr>> textReadouts =
+      text_readouts ? absl::make_optional(server_.stats().textReadouts()) : absl::nullopt;
```

Comment:
This makes an unneeded copy of the vector of textReadouts(). You should either use `absl::optional<std::reference_wrapper<Stats::TextReadoutSharedPtr>>`, or just use `Stats::TextReadoutSharedPtr&` and pass a default-constructed (empty) vector in the case where `text_readouts` is false.

Reply:
Done